src/internal-model.js

```diff
@@ -21,84 +31,91 @@ let InternalModel = Model.extend({
     return _.extend(attributes, computed)
   },
 
+  /**
+   * {Backbone.Model} does not provide "computed" properties.
+   *
+   * @param {string} attr name
+   * @return {mixed|undefined} via {attributes[attr]} or {computed[attr]}
+   */
   get (attr) {
-    if (this.attributes.hasOwnProperty(attr)) {
-      return this.attributes[attr]
-    } else if (this.computed.hasOwnProperty(attr)) {
-      return this.computed[attr].call(this)
-    }
-  },
+    if (this.attributes.hasOwnProperty(attr)) return this.attributes[attr]
 
-  getRelationshipType (relationName) {
-    let relationship = this.getRelationship(relationName)
-    if (_.isArray(relationship.data)) {
-      return 'has-many'
-    } else if (relationship.data) {
-      return 'belongs-to'
-    } else {
-      return 'unknown'
-    }
+    if (this.computed.hasOwnProperty(attr)) return this.computed[attr].call(this)
+
+    return undefined
   },
 
   getRelationshipLink (relationName) {
-    let link = this.getRelationship(relationName).links.related
+    let link = _.result(this.getRelationship(relationName), ['links', 'related'])
+
     if (link == null) {
       throw new Error(`link for, "${relationName}", is undefined for ${this.get('_type')}-${this.id}`)
     }
+
     return link
   },
 
-  getRelationship (relationName) {
+  _getRelationForName (relationName) {
+    return this.relationships && this.relationships[relationName]
+  },
+
+  getRelationship (relationName, strict = true) {
     let modelName = this._getRelationForName(relationName)
```

Comment:
Can we change modelName to a const as well?

Reply:
Done, along with other unmodified `let` definitions in this file.